### ISSUE TITLE
Expose Notes through the API

### ISSUE THREAD

**Issue by freeranger** [st-charles/DSCP-Partner-Portal-API#18]
I don't know if these have any validity on their own - they must have a user and a group so it seems to me they would be exposed via one or the other.
I think via group would be most appropriate and suggest the following:
```
GET /api/groups/:id/notes - returns all notes for that group
GET /api/groups/:id/notes?search=XXXX - searches for notes with matching content
GET /api/groups/:id/notes/:noteid - return a single note
POST /api/groups/:id/notes - create a note for that group
PUT /api/groups/:id/notes/:noteId - update a note
DELETE /api/groups/:id/notes/:noteid - delete a note
```
Not sure if you want to always limit notes to the current user, or at least have the ability to filter by user?  There are a number of ways we could do this, can discuss if this is something you want to do.

For the GET operations, you might want to also return the actual user who created the note rather than their id?  Or of course you can return a link to them.
No need to return a link to the group the note is in because you got to the note by navigating through groups in the first place.

**Comment by commitshappen** [comment]
@freeranger Same with the links, none of the notes need their own resource. They will solely be accessed through a group or user. 

> Not sure if you want to always limit notes to the current user, or at least have the ability to filter by user?

Not sure if this answers your questions, but I'm looking to have the UI pull all the notes related to a group and not just the ones posted by that particular user. See the following UI:
![Imgur](http://i.imgur.com/IJKTjSl.png)

**Comment by freeranger** [comment]
@commitshappen That helps yes.  Looks like this is one of the cases where we will need to expose the created (or updated?) date to show alongside the author.
I think if I leave out the search end point, all the others will cover what you need.
Do you need to return anything about the author apart from their name?

**Comment by commitshappen** [comment]
@freeranger I would expose both created and updated dates, just in case I want to add either or in the future. As of now, however, I'm only planning on utilizing created at.

> Do you need to return anything about the author apart from their name?

Not at this time.

> Also does anything about notes have to be unique or can the same person create as many notes as they like for a given group?

No need to be unique and people can have multiple notes.

> Currently I have no validation for the note text other than it must be present.

I'd also set a minimum text length of 25 characters. Anything less than that shouldn't be logged as a note, as it's not substantial enough to log.

**Comment by freeranger** [comment]
@commitshappen I think this can be closed now?

**Comment by commitshappen** [comment]
Closed by PR #26
